README.md

```diff
@@ -1,49 +1,47 @@
+2. Open `src/researchstudy-mapping.ts`. This file contains a single function: `convertToResearchStudy` function, which handles converting individual trials as returned from the underlying service into FHIR ResearchStudy objects. While the stub is small, the bulk of additional code is likely to be in here.
+3. Open `src/server.ts`. This contains the code to start the service, as well as load configuration data. The default is very simplistic, it simply loads configuration via `configFromEnv()` and passes it off to the `createClinicalTrialLookup()` function. The [dotenv-flow](https://github.com/kerimdzhanov/dotenv-flow) package may be useful for loading configuration information from files.
 
+You may need to further tweek the code stubs provided, and there are other ways to implement a service implemenation. (For example, you could simply extend `ClinicalTrialMatchingService` and pass it a matcher that invokes a method on the generated class.) However the steps above should provide the basic stubs necessary to generate an implementation of a matching service.
 
 # Requirements
 
-The ResearchStudy object passed back by this server must be [FHIR-compliant] (https://www.hl7.org/fhir/researchstudy.html) and satisfy several requirements. 
-Study must contain: 
-- Title 
-- Summary 
+The ResearchStudy object passed back by this server must be [FHIR-compliant] (https://www.hl7.org/fhir/researchstudy.html) and satisfy several requirements.
+Study must contain:
+- Title
+- Summary
 - Site location
-- Phase 
+- Phase
 - Contact Information i.e. sponsor email, phone number
-- Study Type 
+- Study Type
 - Inclusion/ Exclusion criteria
 
-Note: _research-study.ts_ provides a "backup" system for filling in information missing from the object passed back from the matching service. This system fills in the following based on the trial's NctId:
-- Inclusion/Exclusion Criteria
-- Phase 
-- Study Type 
-- Trial Summary
-
 # Running the Server
+
+Note that at present, two environment variables must be set before the server will start: `MATCHING_SERVICE_ENDPOINT` and `MATCHING_SERVICE_AUTH_TOKEN`. If neither are set, the service will raise an error.
+
 1. Run `npm install`
-2. Run `npm run-script build`
-3. Run `npm start`
-4. The service will now be running at http://localhost:3000/
+2. Run `npm start`
+3. The service will now be running at http://localhost:3000/
+
+# Testing
+
+A validation test is provided to validate the ResearchStudy created via this service. Put an example response object in `spec/data/trial_object.json` and this object will be loaded and validated within this test. For this test to run, you must:
+
+1. Have a Java runtime environment installed
+2. [Download a copy of the HL7 validator JAR] (https://storage.googleapis.com/ig-build/org.hl7.fhir.validator.jar) into `spec/data/org.hl7.fhir.validator.jar`
+3. Have properly filled out `spec/data/trial_object.json`
+4. Have implemented the conversion function in `src/researchstudy-mapping.ts`
```

Comment:
This should be updated with the new use of validation.

Reply:
The README has been updated with the new validator.